### DISCUSSION TITLE
MIDI2LR is crashing at start

### DISCUSSION THREAD

**Discussion by krakozavr** [rsjaffe/MIDI2LR#593] [Problems]
Hello.

MIDI2LR is always crashing at LightRoom start.
If I'm closing the error dialog, LR is running but MID2LR still "launched" but it is totally frozen. If I'm force closing and restarting MIDI2LR without restarting Lightroom, it is working but doesn't switching profiles.

If I disabling the plugin before exiting LightRoom, then enabling it after LR restart, everything is working fine - till the next time the plugin will not be disabled. If so, the error is reappearing, and I managed to clear it only by reinstalling MIDI2LR over.

Screenshot with the error box, MIDI2LR log, and system crash log are attached.

MacOS Big Sur, LightRoom 11.1, MIDI2LR 5.2

[log-3.txt](https://github.com/rsjaffe/MIDI2LR/files/7740600/log-3.txt)
[Crash_log-system.txt](https://github.com/rsjaffe/MIDI2LR/files/7740601/Crash_log-system.txt)

![Снимок экрана 2021-12-19 в 12 54 28](https://user-images.githubusercontent.com/16202038/146671083-e1950366-3f46-449f-b5aa-1079e18ae9e1.png)

**Comment by rsjaffe** [comment]
Could you post your MIDI2LR log right after it crashes? ~/Library/Logs/MIDI2LR/MIDI2LR.log

**Comment by krakozavr** [comment]
Unfortunately not, I set it up for my friend, and I have no access to his
computer anymore. I left it on the stage "turn plugin off before exciting
LR, turn it back on after start". It works but annoying.

Best regards,
Alex Potemkin,
AlexArtPhoto.

***@***.***

USA phone: +1 610-714-14-94

My fresh and selected photos: http://yourphoto.nyc
My aerial drone imagery: http://aerial.yourphoto.biz
My studio info and customers's reviews: http://yourphoto.biz
My portfolio on iStockPhoto: http://istock.yourphoto.biz
My portfolio on GettyImages: http://getty.yourphoto.biz
My prints, posters and goods for sale (gifts and home&office wall decor) :
http://alexpotemkin.com

**Comment by rsjaffe** [comment]
Start up Lightroom in Library mode, and see if that fixes the crash.

**Comment by rsjaffe** [comment]
Could you check the latest release (in beta) to see if it fixes it. I was unable to replicate the issue, even with different profiles in each slot, but did identify two potential problems that I've fixed. See https://github.com/rsjaffe/MIDI2LR/discussions/613#discussion-3794305

**Reply by krakozavr** [reply]
I can't check it by myself, I'm in NYC and my friend is in Moscow. But he installed the new version. Now, nothing is works :) the new plugin is starting, and it is connected to lightroom, but doesn't works. I will try to figure out more details but it is complicated remotely.

**Reply by krakozavr** [reply]
OK, some details.
Now, after some "remote shaman's dancing," we have this:
- the plugin is enabled in Plug-In manager
- when LightRoom restarting, Midi2Lr starting accordingly, without crash (so, it us fixed)
- it shows "connected to lightroom, and the plugin is definitely see the controller (the panel is reacting to controller). But nothing is happening in Lightroom.
- when LightRoom closing, the plugin doesn't. And us still indicate " connected to LightRoom" even when Lightroom is not loaded.

He downgraded to v.5.2; it works.

**Comment by rsjaffe** [comment]
Version 5.3 has a bug for macOS users. What you've described is exactly how that bug behaves. I'm not sure what caused this. I suspect it was in an update of a third-party library that introduced it, as I didn't change any code in the section that controls this behavior. I'll have a new test release soon.

**Comment by rsjaffe** [comment]
Could you try [5.3.0.1](https://github.com/rsjaffe/MIDI2LR/releases/tag/v5.3.0.1)? I've rebuilt it with a newer version of the framework. I went through all the changes between 5.2 and 5.3, and nothing should have affected port function. Thinking it's either a permission problem or a third-party library problem. Updated main third-party library and rebuilt.

**Reply by krakozavr** [reply]
did not help. The plugin is starting normally, and it sees the controller but nothing happens in LightRoom.

**Reply by rsjaffe** [reply]
Something is wrong with communication between version 5.3 app and plugin on macOS. I'm still not sure what it is, as I didn't change the code governing the communication! I'm still trying to track this down.

**Comment by rsjaffe** [comment]
Could you try [5.3.0.2](https://github.com/rsjaffe/MIDI2LR/releases/tag/v5.3.0.2)? I've rebuilt the macOS version again.
